### PR TITLE
make M4 test cross platform

### PR DESCRIPTION
**Pull request by dmoody256**
Update M4 test to not pass if the the M4 binary is not found. Also use os.linesep for parsing output so that it works with windows and linux line endings.

## Contributor Checklist:

* [x] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [ ] I have updated `master/src/CHANGES.txt` directory (and read the `README.txt` in that directory)
* [ ] I have updated the appropriate documentation

### CODE REVIEW

**Comment by dmoody256** [comment]
Found this while testing https://github.com/SCons/scons/pull/3270

File: test/M4/M4.py [hunk 1]
```diff
@@ -60,23 +58,16 @@
...
-import sys
-
-if sys.platform == 'win32':
-    # Handle carriage returns.
-    test.must_match(test.workpath('aaa.x'), "line 1\r\nmym4.py\r\nline 3\r\n")
-else:
-    test.must_match(test.workpath('aaa.x'), "line 1\nmym4.py\nline 3\n")
-
-
+test.must_match(test.workpath('aaa.x'), "line 1" + os.linesep + "mym4.py" + os.linesep + "line 3" + os.linesep)
```
Comment (bdbaddog) [inline test/M4/M4.py]: os.linesep.join([....]) ?

File: test/M4/M4.py [hunk 2]
```diff
@@ -87,23 +78,22 @@
...
 """ % locals())
 
-    test.write('foo.x.m4', "line 1\n"
-                           "FFF\n"
-                           "line 3\n")
-
-    test.write('bar.m4', "line 1\n"
-                         "BBB\n"
-                         "line 3\n")
+test.write('foo.x.m4', "line 1" + os.linesep +
```
Comment (bdbaddog) [inline test/M4/M4.py]: same join?

File: test/M4/M4.py (hunk 1 above)
Comment (mwichmann) [reply test/M4/M4.py]: I'd tend to agree, "adding" multiple substrings is usually frowned upon, even though this is "only" a testcase....

Comment (dmoody256) [reply test/M4/M4.py]: updated in dc4b8ba

File: test/M4/M4.py (hunk 2 above)
Comment (dmoody256) [reply test/M4/M4.py]: updated in dc4b8ba

**Comment by dmoody256** [comment]
This one is good to go.

Regarding the appveyor fail, since I only changed the M4.py test code, the other failures must be unrelated. 

As such, the expected mingw failure is there, and the docbook ones look like the vm lost connection, because it passes in the other platforms.

**Pull request merged**